### PR TITLE
Bugfix/fsspec

### PR DESCRIPTION
**Pull request by Mause**

### CODE REVIEW

File: tools/pythonpkg/src/include/duckdb_python/pyfilesystem.hpp [hunk 1]
```diff
@@ -22,23 +22,37 @@ class AbstractFileSystem : public py::object {
 class PythonFileHandle : public FileHandle {
 public:
 	PythonFileHandle(FileSystem &file_system, const string &path, const py::object handle);
+	~PythonFileHandle() {
+		PythonGILWrapper gil;
+		handle.dec_ref();
```
Comment (Mause) [inline tools/pythonpkg/src/include/duckdb_python/pyfilesystem.hpp:27]: TODO: determine if something similar is required in the copy constructor

File: tools/pythonpkg/src/pyfilesystem.cpp [hunk 2]
```diff
@@ -9,6 +9,11 @@ namespace duckdb {
 PythonFileHandle::PythonFileHandle(FileSystem &file_system, const string &path, const py::object handle)
     : FileHandle(file_system, path), handle(handle) {
 }
+PythonFileHandle::~PythonFileHandle() {
+	PythonGILWrapper gil;
+	handle.dec_ref();
+	handle.release();
```
Comment (carlopi) [inline tools/pythonpkg/src/pyfilesystem.cpp:15]: I think this is not needed, even just defining the destructor + calling `Close()` seems to work just fine (but I might be missing some context), the really important things seems to have handle as seekable + adding destructor.

Even just this patch works:
```
diff --git a/tools/pythonpkg/src/include/duckdb_python/pyfilesystem.hpp b/tools/pythonpkg/src/include/duckdb_python/pyfilesystem.hpp
index 572804f075..ddd51e93df 100644
--- a/tools/pythonpkg/src/include/duckdb_python/pyfilesystem.hpp
+++ b/tools/pythonpkg/src/include/duckdb_python/pyfilesystem.hpp
@@ -22,7 +22,11 @@ public:
 class PythonFileHandle : public FileHandle {
 public:
        PythonFileHandle(FileSystem &file_system, const string &path, const py::object handle);
-
+       ~PythonFileHandle() override
+       {
+               PythonFileHandle::Close();
+               //unsure whether specifying the class if needed, seems to work either way
+       }
        void Close() override {
                PythonGILWrapper gil;
                handle.attr("close")();
diff --git a/tools/pythonpkg/src/pyfilesystem.cpp b/tools/pythonpkg/src/pyfilesystem.cpp
index 3aebc2e369..116cc6c000 100644
--- a/tools/pythonpkg/src/pyfilesystem.cpp
+++ b/tools/pythonpkg/src/pyfilesystem.cpp
@@ -31,7 +31,10 @@ unique_ptr<FileHandle> PythonFilesystem::OpenFile(const string &path, uint8_t fl
                throw InvalidInputException("%s: unsupported file flags", GetName());
        }
 
-       const auto &handle = filesystem.attr("open")(py::str(stripPrefix(path)), py::str(flags_s));
+
+       const auto &handle =
+           filesystem.attr("open")(py::str(stripPrefix(path)), py::str(flags_s), py::arg("seekable") = true);
+
        return make_unique<PythonFileHandle>(*this, path, handle);
 }
```

Comment (Mause) [reply tools/pythonpkg/src/pyfilesystem.cpp:15]: The main fix here is obtaining the lock, then running the destructor in the context of the lock (via dec_ref), then making sure the standard object destructor doesn't run by calling release

Comment (Mause) [reply tools/pythonpkg/src/pyfilesystem.cpp:15]: (calling close doesn't do any of that)

**Pull request merged**